core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Index.java

```diff
@@ -82,26 +83,29 @@ public Index(Path path, Index.Type type, Factory<T> factory) {
   /** Starts the reading and index process. */
   public void index() {
     items.clear();
-    try (BufferedReader br = Files.newBufferedReader(this.path, UTF_8)) {
-      String line = br.readLine();
-      if (line != null) {
-        line = br.readLine();
-      }
-      while (line != null) {
-        T item = factory.build(line.split("\t"));
-        total++;
-        int hash;
-        if (type.equals(Index.Type.BY_CLASS)) {
-          hash = Objects.hash(item.encClass());
-        } else {
-          hash = Objects.hash(item.encClass(), item.encMethod());
-        }
-        items.put(hash, item);
-        line = br.readLine();
-      }
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+    paths.forEach(
+        path -> {
+          try (BufferedReader br = Files.newBufferedReader(path, UTF_8)) {
+            String line = br.readLine();
+            if (line != null) {
+              line = br.readLine();
```

Comment:
Please add a comment about what's happening here. Presumably this skipping the TSV headers, right?

Reply:
f490c9278abb1fd31da6b8d56bd12aad4bb80e0a